shell/platform/android/io/flutter/plugin/platform/VirtualDisplayController.java

```diff
@@ -145,15 +146,22 @@ public void run() {
           public void onViewDetachedFromWindow(View v) {}
         });
 
-    presentation =
+    // Initialize a new presentation. We don't want to cancel
+    // the old one before we can properly show the new one to
+    // take over where it left off.
```

Comment:
I'm not sure I understand this comment, what side effect does `newPresentation.show()` has that makes `presentation.cancel()` safer?